aws/defaults/handlers.go

```diff
@@ -204,10 +206,25 @@ var RetryMetricHeaderHandler = aws.NamedHandler{
 		if max := r.Retryer.MaxAttempts(); max != 0 {
 			parts = append(parts, fmt.Sprintf("max=%d", max))
 		}
-		// TODO "ttl=YYYYmmddTHHMMSSZ"
-		//    ttl = current_time + socket_read_timeout + estimated_skew
-		// SDK doesn't implement clock skew yet and not obvious how to obtain
-		// read deadlines.
+
+		type timeoutGetter interface {
+			GetTimeout() time.Duration
+		}
+
+		// Attempt extract the TTL from context deadline, or timeout on the client.
+		var ttl time.Time
+		if v, ok := r.Config.HTTPClient.(timeoutGetter); ok && v.GetTimeout() > 0 {
+			ttl = sdk.NowTime().Add(v.GetTimeout())
```

Comment:
`GetTimeOut()` should be called just once.

Reply:
In this case since the else condition needs to be invoked if there is no timeout, the else has to be duplicated, or the if becomes more complicated if the `GetTimeout` call is only called once.